packages/deployment/test/test_adapter.js

```diff
@@ -26,51 +53,46 @@ test('downloaded latest version of the message adapter', async (t) => {
     t.context.dest
   );
 
-  // confirm the zip file exist
-  const dirExists = fs.statSync(unzipped);
-  t.truthy(dirExists);
+  t.truthy(await pStat(unzipped));
+
   t.is(unzipped, t.context.dest);
 
-  // confirm the extracted folder exists
   t.true(await fs.pathExists(t.context.src));
 });
 
-test('should download specific version if provided', async (t) => {
-  const version = 'v1.0.0';
-  // create temp directory
-  const unzipped = await fetchMessageAdapter(
-    version,
+test('fetchMessageAdapter() can download a specific version if provided', async (t) => {
+  const scope = nock('https://github.com')
+    .get('/nasa/cumulus-message-adapter/releases/download/v1.0.0/cumulus-message-adapter.zip')
+    .replyWithFile(200, './test/fixtures/zipfile-fixture.zip');
+
+  await fetchMessageAdapter(
+    'v1.0.0',
     gitPath,
     'cumulus-message-adapter.zip',
     t.context.src,
     t.context.dest
   );
-  // confirm the zip file exist
-  const dirExists = fs.statSync(unzipped);
-  t.truthy(dirExists);
-  t.is(unzipped, t.context.dest);
 
-  // confirm version number
-  const versionFile = path.join(t.context.dest, 'message_adapter/version.py');
-  const versionString = fs.readFileSync(versionFile).toString();
-  const rTest = /v[\d]\.[\d]\.[\d]/;
-  t.true(rTest.test(versionString));
+  t.true(scope.isDone());
```

Comment:
What we're testing here is that, if we ask `fetchMessageAdapter()` to fetch a particular version of the message adapter, `fetchMessageAdapter()` will request the correct version from Github.  

Before this change, the only way to verify that the correct version had been fetched was to get the zip file, open it up, and see if the version file was correct.  After this change, we can assert that the correct request was sent, and don't have to depend on the contents of the zip file for that verification.

Reply:
If I'm understanding, it seems like the previous version of the tests was doing both though: Verifying that `fetchMessageAdapter` requests the correct version and that the downloaded file has the correct version. Checking the validity of the request and the response. Maybe testing that the downloaded file itself has the correct version is superfluous since if it doesn't that's probably an error on our part, but just flagging it

Reply:
Correct, the previous version was also validating the contents of the zip file that was fetched from Github.  The contents of that file have nothing to do with whether the `fetchMessageAdapter` function is working properly, though.  If we screw up and deploy the message adapter using the wrong version, that isn't a reflection on whether this code is correct or not.

Reply:
Put another way, this is a unit test for the `fetchMessageAdapter` function.  If we put an incorrect version of the message adapter on Github, this test will fail, but there will be nothing that can be fixed in the `fetchMessageAdapter` function that will make the test pass.  So, assuming that the correct file was requested from Github (which we are testing here), testing the content of that file doesn't tell us anything about the correctness of the `fetchMessageAdapter` function.

Reply:
Yes, I understand that. It just seems like the previous version was more than just a unit test for `fetchMessageAdapter` and I was wondering if we wanted to keep that test. But I think we're on the same page

Reply:
👍